### DISCUSSION TITLE
Searx Primer: A theme based on GitHub's design language, Primer.

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#962] [Show and tell]
I published a theme for SearXNG that takes heavy inspiration from Primer, GitHub's design language.

The theme is not originally mine; I found it a while ago but it seems to have been wiped from the internet, so I guess I'm the new maintainer now :P

This theme is just a userstyle; it's not like an actual theme, if that makes sense. If anyone wants to turn this into an actual theme that would be really cool, but I don't know if that could easily be done.

**https://github.com/lx-is/searx-primer**

**Light mode**
![light@0 5x](https://user-images.githubusercontent.com/100533118/158035536-716ab377-5295-4f40-baf8-c4e6f137ceb1.png)

**Dark mode**
![dark@0 5x](https://user-images.githubusercontent.com/100533118/158035554-b722a51e-d620-4e04-a800-a162ec21b5f1.png)

**Comment by mrpaulblack** [comment]
Hi,
This is a really cool theme. It looks like it is based on oscar tho, so just an an FYI we are going to discontinue oscar and remove it from SearXNG core eventually... (This is not going to happen today or tomorrow, but eventually in a few months maybe). You can port your theme to Searx oscar or to our SearXNG simple theme... (Check out simple theme CSS source here: https://github.com/searxng/searxng/tree/master/searx/static/themes/simple/src/less).

Thanks for your great work with that theme :+1:

**Reply by ghost** [reply]
Thanks for the heads up. If that's the case I may try to port the design to the simple theme.

If I may ask, what's the reason behind removing Oscar? It seems a lot of users prefer it, myself included.

**Reply by mrpaulblack** [reply]
The reasoning is that oscar is based on bootstrap 3 which is deprecated and its not arbitrary to upgrade to bootstrap 5. (We would first need to upgrade to bootstrap 4 and rewrite part of the templates and then upgrade to bootstrap 5)

The simple theme does not use any HTML frameworks like that and is easier to maintain and also uses more modern CSS approaches for stuff like the image grid for example...
We already put a lot of work over the last few months making the simple theme more modern looking (for example the auto dark theme) and since its also a lot smaller in production we decided to drop oscar (if you use dev tools you can see all the CSS and JS with oscar and how the simple theme is much smaller and needs less JS for the same feature set)

That is the reasoning behind dropping oscar (also the simple theme looks better IMO, but that is personal preference...). In the end the simple theme is much easier to maintain and maintaining 2 themes at the same time with the small dev team on SearXNG core is not feasible IMO.

**Reply by ghost** [reply]
I see, that makes sense; thanks for explaining.

**Comment by return42** [comment]
> (This is not going to happen today or tomorrow, but eventually in a few months maybe). 

tbh I wanted to prepare a DRAFT the next days removing oscar ..

**Comment by mrpaulblack** [comment]
Oh ok yeah please do -> its probably coming down to https://github.com/searxng/searxng/pull/894 getting merged. After that I see no point in keeping oscar :+1: 

Will do the review late this week for that PR...

**Reply by dalf** [reply]
> After that I see no point in keeping oscar 👍

The high contrast settings has issues with FF & Chrome:
* https://github.com/searxng/searxng/issues/951
* https://dev.searxng.org/screenshots.html#config=Chrome+1920+-+high+contrast&test_case=preferences

I think it can be fixed quickly.

**Reply by mrpaulblack** [reply]
Well the fix is the Aria PR is it not?

**Reply by dalf** [reply]
* Nope, the Aria PR fixes various issues with screen readers (it doesn't change the CSS)
* The fix to the issue 951 is some CSS changes:
  * either by changing how the components are displayed (`<select>` element)
  * either using dedicated CSS rules ( https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors )

**Reply by mrpaulblack** [reply]
hmm ok mb.
